core/lib/src/config/config.rs

```diff
@@ -577,6 +578,56 @@ impl Config {
         Ok(())
     }
 
+    /// Sets the TLS configuration in `self`.
+    ///
+    /// Certificates are read from `certs_read`. The certificate chain must be
+    /// in X.509 PEM format. The private key is read from `key_read`. The
+    /// private key must be an RSA key in either PKCS#1 or PKCS#8 PEM format.
+    ///
+    /// # Errors
+    ///
+    /// If reading either the certificates or private key fails, an error of
+    /// variant `Io` is returned. If either the certificates or private key
+    /// files are malformed or cannot be parsed, an error of `BadType` is
+    /// returned.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rocket::config::Config;
+    /// use std::io::Cursor;
+    ///
+    /// # use rocket::config::ConfigError;
+    /// # fn config_test() -> Result<(), ConfigError> {
+    /// let mut config = Config::development()?;
+    /// let certfile_bytes = Cursor::new([10, 10, 10, 10].to_vec());
+    /// let keyfile_bytes = Cursor::new([10, 10, 10, 10].to_vec());
+    /// config.set_tls_from_read(certfile_bytes, keyfile_bytes)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "tls")]
+    pub fn set_tls_from_read<R: Read + Seek>(&mut self, certs_read: R, key_read: R) -> Result<()> {
```

Comment:
(bikeshed) IMO names like `certs_data` and `key_data` are more natural `certs_read`.